### DISCUSSION TITLE
request --prepare option for setup

### DISCUSSION THREAD

**Discussion by petiepooo** [Security-Onion-Solutions/securityonion#7975] [Ideas] [answered]
I would love to have an option during setup that just installs the dependent OS packages, salt, and docker packages a full run of so-setup will need, then downloads all so-* images from ghcr.io/security-onion-solutions, and then exits without any further system or salt configuration.

This would be useful for air-gapped installs as well as situations where a system is prepared in a central location with higher bandwidth, then deployed to a location with limited bandwidth.  If not an option to so-setup, even just a list of apt/yum packages (maintained as they change) and pinned salt/docker versions for the current and past few releases would help.  That would be easier than digging through the setup files to locate all the installed packages and versions.

**Comment by dougburks** [comment, marked as answer]
It looks like you've already requested this here:
https://github.com/Security-Onion-Solutions/securityonion/issues/6835

**Comment by petiepooo** [comment]
Oh, sure enough.  It seems I've been looking for the capability for quite some time now.  I'll see if I can put together a patch and a pull request if you think it might be something you'd consider including.
